specs/phase0/beacon-chain.md

```diff
@@ -1675,7 +1675,7 @@ def process_block_header(state: BeaconState, block: BeaconBlock) -> None:
         slot=block.slot,
         proposer_index=block.proposer_index,
         parent_root=block.parent_root,
-        state_root=Bytes32(),  # Overwritten in the next process_slot call
+        state_root=Root(),  # Overwritten in the next process_slot call
```

Comment:
We tend to use `Bytes32` instead of `Root()` or `Hash()` to avoid confusion with a hash-tree-root or hash operation (`hash()` may be interpreted as a sha-256 of an empty input). Alternatively we could use a `ZERO_ROOT` constant or something like that.

Reply:
Not sure about the `ZERO_ROOT` actually, we should not change the phase0 specs for smaller things like this. `Bytes32` isn't bad. Which type checker did you use?

Reply:
Thanx! I like the `ZERO_ROOT` idea!

Reply:
> Not sure about the ZERO_ROOT actually, we should not change the phase0 specs for smaller things like this. 

This is reasonable, definitely! Haven't thought about that. I'll revert then.

> `Bytes32` isn't bad.

It's not bad, but semantics become a bit complicated.
`state_root` has `Root` type, so assigning `Bytes32` value to it means implicit conversion.
Actually, I believe I have to model implicit conversions anyway.

> Which type checker did you use?

I'm using a custom type checker. It's based on PEP-484, of course, but I'm more flexible with my tool. E.g. can introduce my own typing rules to match the pyspecs better.
My goal eventually is to define formal semantics for pyspecs, i.e. to a subset of Python, used to express the pyspecs.

Reply:
Reverted back to `state_root=Bytes32()`